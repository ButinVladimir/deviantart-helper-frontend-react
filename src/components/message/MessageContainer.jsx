import { connect } from 'react-redux';
import Message from './Message';
import hideMessageActionCreator from '../../redux/action-creators/shared/hide-message';

/**
 * @description
 * Maps Redux state to header props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  showMessage: state.shared.showMessage,
  message: state.shared.message,
});

/**
 * @description
 * Maps Redux dispatch to header props.
 *
 * @param {Function} dispatch - The dispatch.
 * @returns {Object} Props.
 */
const mapDispatchToProps = dispatch => ({
  hideMessageHandler: () => dispatch(hideMessageActionCreator()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Message);
