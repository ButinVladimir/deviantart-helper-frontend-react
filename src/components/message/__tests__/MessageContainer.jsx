import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import MessageContainer from '../MessageContainer';
import createDefaultState from '../../../redux/states/state';
import hideMessageActionCreator from '../../../redux/action-creators/shared/hide-message';

jest.mock('../../../redux/action-creators/shared/hide-message', () => jest.fn(() => ({ type: 'TEST_ACTION' })));

describe('MessageContainer', () => {
  it('can be rendered correctly', () => {
    const state = {
      ...createDefaultState(),
      shared: {
        message: 'Message text',
        messageColor: 'info',
        showMessage: true,
      },
    };

    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const wrapper = mount(
      <Provider store={store}>
        <MessageContainer />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can hide message', () => {
    const state = {
      ...createDefaultState(),
      shared: {
        message: 'Message text',
        messageColor: 'info',
        showMessage: true,
      },
    };

    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const wrapper = mount(
      <Provider store={store}>
        <MessageContainer />
      </Provider>,
    );

    wrapper.find('button#hide-message-button').simulate('click');
    const actions = store.getActions();

    expect(hideMessageActionCreator).toHaveBeenCalled();
    expect(actions).toEqual([
      { type: 'TEST_ACTION' },
    ]);
  });
});
