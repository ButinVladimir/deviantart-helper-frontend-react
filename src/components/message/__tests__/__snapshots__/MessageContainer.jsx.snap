// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContainer renders correctly 1`] = `
<section
  className="section"
  style={Object {}}
>
  <div
    className="container is-fullhd"
    style={Object {}}
  >
    <div
      className="notification is-info"
      style={Object {}}
    >
      <button
        className="delete"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        tabIndex={0}
      >
        Hide
      </button>
      Message text
    </div>
  </div>
</section>
`;
